### PR TITLE
Add support article for advanced search and images

### LINKED ISSUE #61: Support page for advanced search
Based on conversation here: https://chat.kobotoolbox.org/#narrow/stream/4-KoBo-Dev/topic/Public.20collections.20final.20final.20topic/near/19088 and release of public collections.

### PR DESCRIPTION
**Pull request by joshuaberetta**
Support article for advanced searching in public collections.

## Related issues
closes #61

### CODE REVIEW

File: source/public_collections_advanced_search.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Public Collections Advanced Search
+
+**_Note that the user-friendliness of advanced searching will improve in the
+coming months._**
```
Comment (jnm) [inline source/public_collections_advanced_search.md:4]: Let's not give this time frame as I'm not sure we'll be able to stick to it. Perhaps
> Please note that the search capability is a work in progress, and we plan to add a more user-friendly syntax in future releases.

File: source/public_collections_advanced_search.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+### Valid Search Field Operators
+
+The field operator is the value after the last double-underscore in the search
+field, i.e. `__icontains`.
+
+- For _case-sensitive_ **text** searches, the following operators can be used:
+  `contains`, `exact`, `startswith`
+- For _case-insensitive_ **text** searches: `icontains`, `iexact`, `istartswith`
+- For **numeric** search, the following operators are valid: `lte`, `gte`
```
Comment (jnm) [inline source/public_collections_advanced_search.md:33]: `exact` works for numeric fields (really any value), plus there are `lt` and `gt`

File: source/public_collections_advanced_search.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+**_Note that the user-friendliness of advanced searching will improve in the
+coming months._**
+
+### The Default Search Behaviour
+
+When you enter a term into the search bar, the following fields are searched by
+default when no explicit field is given (combined with the `OR` operator):
+`name__icontains`, `owner__username__icontains`,
+`settings__description__icontains`, `summary__icontains`,
+`tags__name__icontains`, `uid__icontains`.
```
Comment (jnm) [inline source/public_collections_advanced_search.md:12]: I think this is great for power users (maybe the target audience of this article?), but I can imagine most people better understanding something less jargon-y like
> When you enter a term into the search bar without specifying a field, your query will return results where that term, regardless of capitalization, can be found in:
> * The name of the survey, collection, question, block, or template;
> * The owner's username;
> * The description;
> * The summary, which contains all question labels; [and maybe other stuff? I don't remember]
> * The name of any assigned tag;
> * The object's UID.

**Review by jnm — CHANGES_REQUESTED** [review]

File: source/public_collections_advanced_search.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+### The Default Search Behaviour
+
+When you enter a term into the search bar without specifying a field, your
+query will return results where that term, regardless of capitalization, can be
+found in:
+- The name of the survey, collection, question, block, or template;
+- The owner's username;
+- The description;
+- The summary, which contains all question labels; [and maybe other stuff? I
+  don't remember]
```
Comment (jnm) [inline source/public_collections_advanced_search.md:15]: \*giggle\* replace my stuff between the brackets with a more complete description of summary, or just trash it if time doesn't permit :smile:

**Review by jnm — CHANGES_REQUESTED** [review]

File: source/public_collections_advanced_search.md (hunk 4 above)
Comment (joshuaberetta) [reply source/public_collections_advanced_search.md:15]: 😂 @jnm I was clearly paying attention while doing this

File: source/public_collections_advanced_search.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+settings__description__icontains:examples OR summary__icontains:examples OR
+tags__name__icontains:examples OR uid__icontains:examples
+```
+
+![examples](/images/public_collections_advanced_search/advanced_search_1.png)
+
+### Valid Search Field Operators
+
+The field operator is the value after the last double-underscore in the search
+field, i.e. `__icontains`.
```
Comment (jnm) [inline source/public_collections_advanced_search.md:32]: I'm feeling that this should be "e.g." ("exempli gratia" / "for example"), but I guess if it's referring to the query underneath the previous heading, maybe "i.e." ("id est" / "that is") is correct.
image

**Review by jnm — APPROVED** [review]
I think this is good to go; thanks for working on it. Future work could be improving the screenshots so that the set of results illustrates what the different filters are doing. I'd consider it to be @Kalyan-Lama's decision as to how something like that should be prioritized.

File: source/public_collections_advanced_search.md (hunk 5 above)
Comment (joshuaberetta) [reply source/public_collections_advanced_search.md:32]: @jnm I always get confused which of those is best to use when 😂 should I just leave it as is then?

**Pull request merged**